### PR TITLE
split nix.conf man page into sections

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
# Motivation

I have often run into looking up the command line flag syntax for settings, and it was hard to find the important bit. Splitting it up into sections makes the document easier to scan.

# Checklist for maintainers

Maintainers: tick if completed or explain if not relevant

 - [ ] agreed on idea
 - [ ] agreed on implementation strategy
 - [ ] tests, as appropriate
   - functional tests - `tests/**.sh`
   - unit tests - `src/*/tests`
   - integration tests - `tests/nixos/*`
 - [ ] documentation in the manual
 - [ ] documentation in the internal API docs
 - [ ] code and comments are self-explanatory
 - [ ] commit message explains why the change was made
 - [ ] new feature or incompatible change: updated release notes

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: doc/manual/src/command-ref/conf-file-prefix.md [hunk 1]
```diff
@@ -4,49 +4,57 @@
...
 
-A configuration setting usually overrides any previous value. However,
-you can prefix the name of the setting by `extra-` to *append* to the
-previous value. For instance,
+## Command line flags
 
-    substituters = a b
-    extra-substituters = c d
+Every configuration setting has a corresponding command line flag (e.g. `--max-jobs 16`).
+Boolean settings do not need an argument, and can be explicitly disabled with the `no-` prefix (e.g. `--keep-failed` and `--no-keep-failed`).
```
Comment (edolstra) [inline doc/manual/src/command-ref/conf-file-prefix.md:55]: It looks like this removes the mention of `--option`. Was that intentional? Either way is fine with me.

**Review by edolstra — APPROVED** [review]

File: doc/manual/src/command-ref/conf-file-prefix.md (hunk 1 above)
Comment (Ericson2314) [inline doc/manual/src/command-ref/conf-file-prefix.md:55]:
```suggestion

There are two ways to set options on the CLI, with a generic `--option` flag, and with option-specific flags.

The generic `--option` way takes two arguments: `--option <name> <value>`.
It is interpreted exactly like a `<name> = <value>` in a setting file.
(In particular `extra-` is supported and unknown options names are warnings.)

Every configuration setting has corresponding command line flag (e.g. `--max-jobs 16`).
Boolean settings do not need an argument, and can be explicitly disabled with the `no-` prefix (e.g. `--keep-failed` and `--no-keep-failed`).
Unknown option names are invalid flags (unless there is already a flag with that name) so unlike with `--option` those become errors not warning.
```

Comment (fricklerhandwerk) [reply doc/manual/src/command-ref/conf-file-prefix.md:56]: Alright, I'll think about how to put this cleanly.

Comment (fricklerhandwerk) [reply doc/manual/src/command-ref/conf-file-prefix.md:56]: It does, and it was intentional, but it's probably not a good idea removing it.

**Pull request merged**